src/main/java/io/vertx/ext/shell/command/base/VerticleDeploy.java

```diff
@@ -50,16 +52,39 @@
 public class VerticleDeploy extends AnnotatedCommand {
 
   private String name;
+  private String options;
+  private DeploymentOptions deploymentOptions;
 
   @Argument(index = 0, argName = "name")
   @Description("the verticle name")
   public void setName(String name) {
     this.name = name;
   }
 
+  @Argument(index = 1, argName = "options", required = false)
+  @Description("the verticle deployment options as JSON string")
+  public void setOptions(String options) {
+    this.options = options;
+  }
+
+  protected DeploymentOptions getDeploymentOptions() {
+    return (options != null && !options.isEmpty()) ?
+      new DeploymentOptions(new JsonObject(options)) : new DeploymentOptions();
+  }
+
   @Override
   public void process(CommandProcess process) {
-    process.vertx().deployVerticle(name, ar -> {
+    try {
+      deploymentOptions = getDeploymentOptions();
+    } catch (RuntimeException e) {
+      process.write("Could not deploy " + name + " cause of invalid JSON string as a deployment options\n");
```

Comment:
I think the message should be more generic to the end user because the details will be provided from written stack trace 

```javascript
process.write("Could not deploy " + name + " with deployment options\n");
```